### PR TITLE
added package to download files from urls

### PR DESCRIPTION
**Pull request by gogogadgetbioprocess**

### CODE REVIEW

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	}
+	defer f.Close()
+
+	if _, err := io.Copy(f, res.Body); err != nil {
+		return nil, err
+	}
+
+	f.Close()
+
+	fmt.Println("made file: ", filename)
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: remove gratuitous output

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+// get a file from a url link, the file will be given the specified filename
+func UrlToFile(url string, filename string) (f *os.File, err error) {
+
+	fmt.Println("getting url: ", url)
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: remove gratuitous output

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+// get a file from a url link, the file will be given the specified filename
+func UrlToFile(url string, filename string) (f *os.File, err error) {
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: Probably not right to return os.File if you are going to close the file

Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: What other things do you intend to download? Would `download.File(url string)` be sufficient?

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// package for downloading files
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: Go style is that package comments should begin:
```go
// Package download ...
```

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+package download
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+// get a file from a url link, the file will be given the specified filename
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: Godoc comments should be complete sentences.
```go
// UrlToFile downloads a url to the given filename. 
```

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go (hunk 3 above)
Comment (gogogadgetbioprocess) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: so it should just return an error?

**Comment by gogogadgetbioprocess** [comment]
Can this get merged in now?

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 6]
```diff
@@ -1,38 +1,33 @@
-// package for downloading files
+// package download, for downloading files
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: Doc comments should be complete sentences

  https://golang.org/doc/effective_go.html#commentary

```go
// Package download provides convenience functions for downloading files
```

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 7]
```diff
@@ -1,38 +1,33 @@
...
 	"io"
 	"net/http"
 	"os"
 )
 
-// get a file from a url link, the file will be given the specified filename
-func UrlToFile(url string, filename string) (f *os.File, err error) {
-
-	fmt.Println("getting url: ", url)
+// func File downloads a url to the given filename.
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: File downloads the data at a url to the given filename. If there is an error, the file will contain the partially downloaded data.

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 8]
```diff
@@ -1,38 +1,33 @@
...
+// On error, the file will be left behind.
+func File(url string, filename string) (err error) {
 
+	var f *os.File
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: delete

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 9]
```diff
@@ -1,38 +1,33 @@
...
 	res, err := http.Get(url)
 
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer res.Body.Close()
 
 	f, err = os.Create(filename)
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: :=

**Comment by gogogadgetbioprocess** [comment]
ok, made those changes now

**Pull request merged**